report/5.Implementation/MovementModule.tex

```diff
@@ -0,0 +1,117 @@
+\section{Movement module}
+The movement module is responsible for making the turret move according to a target.
+The module consists of multiple methods which is responsible for minor areas of the overall function. 
+In the following the movement module will be elaborated upon and some of the main methods will be explained. 
+
+\subsection{The motors}
+Before using the Lego motors the NXT first needs to know which ports are used to each motor. 
+This is done as follows:
+\begin{lstlisting}[language=CSharp, label={lst:DeviceInit},caption={ecrobot\_device\_initialize method from movement.c}]
+void ecrobot_device_initialize(void){
+  init_motor(NXT_PORT_A, 'x', 20);
+  init_motor(NXT_PORT_B, 'y', 20);
+}
+\end{lstlisting}
+As it ca be seen on Snippet~\ref{lst:InitMotor} the port on the NXT is specified with an orientation and an initial speed.
+The \textit{init\_motor} method is declared as follows where the different parameters of the motor function is set up with initial values.
+
+\begin{lstlisting}[language=CSharp,label={lst:InitMotor},caption={init\_motor method from movement.c}]
+bool init_motor(uint8_t motor_id, char orientation, uint16_t speed){
+  if(orientation == 'x'){
+    x_motor = motor_id;
+    x_motor_speed = speed;
+    ecrobot_set_motor_speed(x_motor, 0);
+    nxt_motor_set_count(x_motor, 0);
+    current_location.x = ecrobot_get_motor_rev(motor_id);
+    return true;
+  }
+  else if(orientation == 'y'){
+    y_motor = motor_id;
+    y_motor_speed = speed;
+    ecrobot_set_motor_speed(y_motor, 0);
+    nxt_motor_set_count(y_motor, 0);
+    current_location.y = ecrobot_get_motor_rev(motor_id);
+    return true;
+  }
+  return false;
+}
+\end{lstlisting}
+As it can be seen on Snippet~\ref{lst:DeviceInit} line 2 and 3 the parameter for the motor\_id is the port id on the NXT device. 
```

Comment:
```suggestion
As it can be seen on Snippet~\ref{lst:DeviceInit} lines 2 and 3, the parameter for the motor\_id is the port id on the NXT device. 
```